inference-engine/tests/ie_test_utils/common_test_utils/file_utils.hpp

```diff
@@ -115,4 +116,67 @@ inline bool directoryExists(const std::string &path) {
     return false;
 }
 
+inline void directoryFileListRecursive(const std::string& name, std::vector<std::string>& file_list) {
+    struct CloseDir {
+        void operator()(DIR* d) const noexcept {
+            if (d) {
+                closedir(d);
+            }
+        }
+    };
+    using Dir = std::unique_ptr<DIR, CloseDir>;
+    Dir directory(opendir(name.c_str()));
+    struct dirent *entire;
+    if (directory) {
+        const std::string current_dir{"."};
+        const std::string parent_dir{".."};
+        while ((entire = readdir(directory.get())) != nullptr) {
+            if (std::string(entire->d_name) == parent_dir || std::string(entire->d_name) == current_dir) {
+                continue;
+            }
+            std::string path = name + CommonTestUtils::FileSeparator + entire->d_name;
+            if (directoryExists(path)) {
```

Comment:
Do we expect some symlink loops / max search depth?

Reply:
For now, no such cases, but it's a good comment, It's really undesired to fall into infinite loop.
I've created a corresponding  issue 50291, to fix it further.